docs/fastdds/transport/datasharing.rst

```diff
@@ -0,0 +1,213 @@
+    it differs from :ref:`transport_sharedMemory_sharedMemory`
+    in that :ref:`transport_sharedMemory_sharedMemory` is a full-compliant transport.
+    That means that with :ref:`transport_sharedMemory_sharedMemory`
+    the data being transmitted must be copied from the DataWriter history to the transport
+    and from the transport to the DataReader.
+    With Data-sharing these copies can be avoided
+
+The figure below shows a comparison between the different transports available in *Fast DDS*.
+
+.. figure:: /01-figures/fast_dds/transport/transport_comparison.png
+    :align: center
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Overview
+---------
+
+When the DataWriter is created, *Fast DDS* will pre-allocate a pool of
+|ResourceLimitsQosPolicy::max_samples-api| + |ResourceLimitsQosPolicy::extra_samples-api| samples that reside
+in a shared memory mapped file.
+When publishing new data, the DataWriter will take a sample from this pool and add it to its history,
+and notify the DataReader which sample from the pool has the new data.
+
+The DataReader will have access to the same shared memory mapped file,
+and will be able to access the data published by the DataWriter.
+
+
+Constraints
+-----------
+
+This feature is available only if the following requirements are met:
+
+* The |DataWriter| and |DataReader| have access to the same shared memory.
+* The |Topic| has a bounded |TopicDataType|,
+  i.e., its |TopicDataType::is_bounded-api| member function returns true.
+* The DataWriter is configured with |PREALLOCATED_MEMORY_MODE-api| or |PREALLOCATED_WITH_REALLOC_MEMORY_MODE-api|.
+
+
+Data-sharing delivery configuration
+-----------------------------------
+
+Data-sharing delivery can be configured in the |DataWriter| and the |DataReader|
+using :ref:`datasharingqospolicy`.
+Four attributes can be configured:
+
+* The data-sharing delivery kind
+* The shared memory directory
+* The data-sharing domain identifiers.
+* The maximum number of data-sharing domain identifiers.
+
+Data-Sharing delivery kind
+""""""""""""""""""""""""""
+
+Can be set to one of three modes:
+
+* **AUTO**: If both a DataWriter and DataReader meet the requirements,
+  data-sharing delivery will be used between them.
+  This is the default value.
+* **ON**: Like **AUTO**, but the creation of the entity will fail if the requirements are not met.
+* **OFF**: No data-sharing delivery will be used on this entity.
+
+The following matrix shows when two entities are data-sharing compatible according to their configuration
+(given that the entity creation does not fail and that both entities have access to a shared memory):
+
+.. |common_ids| replace:: Only if they have common domain IDs
+.. |bounded_and_common_ids| replace:: Only if the TopicDataType is bounded and they have common domain IDs
+
++------+------------------+-------------------+--------------------------+
+|      | ON               | OFF               | AUTO                     |
```

Comment:
NIT: Is there a way to add DataWriter and DataReader labels to each table dimension? It would make it more readable.

Reply:
Not easy to do beautifully in sphinx. Repeating `READER` and `WRITER` on each header also looks strange. I left it as I did because at least the table is symmetric, and it doesn't matter which one is a reader/writer

Reply:
Perhaps with a list-table?